src/react-native-animated-charts/src/charts/linear/ChartLines.tsx

```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+import Animated, {
+  useAnimatedStyle,
+  withTiming,
+} from 'react-native-reanimated';
+import { Line, LineProps, Svg } from 'react-native-svg';
+import { useChartData } from '../../helpers/useChartData';
+import withReanimatedFallback from '../../helpers/withReanimatedFallback';
+import { FIX_CLIPPED_PATH_MAGIC_NUMBER } from './ChartPath';
+
+interface ChartLineProps extends LineProps {
+  color?: string;
+  length: number;
+  thickness: number;
+}
+
+function ChartLineFactory(orientation: 'horizontal' | 'vertical') {
+  const isVertical = orientation === 'vertical';
+  const ChartLine: React.FC<ChartLineProps> = ({
+    color = '#000000',
+    thickness = 2,
+    length,
+    ...props
+  }) => {
+    const { positionX, dotScale, currentPath, height } = useChartData();
+
+    const currentPositionVerticalLineStyle = useAnimatedStyle(
+      () => ({
+        opacity: dotScale.value,
+        transform: [{ translateX: positionX.value }],
+      }),
+      []
+    );
+
+    const openingPositionHorizontalLineStyle = useAnimatedStyle(() => {
+      const firstPoint = currentPath?.points?.[0];
+      return {
+        opacity: currentPath?.points.length === 0 ? 0 : 1,
+        transform: [
+          {
+            translateY: withTiming(
+              firstPoint ? firstPoint?.y * height + 10 : 0
```

Comment:
```suggestion
              (firstPoint?.y ?? 0) * height + 10 
```

typically, we use optional chaining, not ternary

Reply:
This code makes me overthink it and I don't like it. Ternary operator makes it way easier and obvious to understand what's going on here.